recipes/libcurl/all/patches/005-FindNGHTTP2.patch

```diff
@@ -0,0 +1,24 @@
+diff --git a/CMake/FindNGHTTP2.cmake b/CMake/FindNGHTTP2.cmake
+--- a/CMake/FindNGHTTP2.cmake 
++++ b/CMake/FindNGHTTP2.cmake
+@@ -21,6 +21,14 @@
+ ###########################################################################
+ include(FindPackageHandleStandardArgs)
+ 
++find_package(libnghttp2)
++
++if(libnghttp2_FOUND)
++
++add_library(NGHTTP2 ALIAS libnghttp2::libnghttp2)
++
++else()
++
+ find_path(NGHTTP2_INCLUDE_DIR "nghttp2/nghttp2.h")
+ 
+ find_library(NGHTTP2_LIBRARY NAMES nghttp2)
+@@ -37,3 +45,4 @@
+ set(NGHTTP2_LIBRARIES ${NGHTTP2_LIBRARY})
+ 
+ mark_as_advanced(NGHTTP2_INCLUDE_DIRS NGHTTP2_LIBRARIES)
++endif()
+\ No newline at end of file
```

Comment:
Fragile and very dependent to lib dir injection by conan_basic_setup(). It will break with CMakeToolchain in conan v2.

I don't understand how it can fix anything. libcurl doesn't use a `NGHTTP2` target.

Reply:
@SpaceIm if the option  with_nghttp2=True in the libcurl ist set, the libcurl depends on libnghttp2 (see conanfile.py) and the package **NGHTTP2** is needed and the variables **NGHTTP2_INCLUDE_DIRS** and **NGHTTP2_LIBRARIES** are used (CMakeList.txt LL 449). So I think the libcurl does use the NGHTTP2. I can't judge if this will break with conan v2. But what ist the alternative?

Reply:
@SpaceIm If you builf the libcurl package with the option with_nghttp2=True the variables **NGHTTP2_INCLUDE_DIRS ** and **NGHTTP2_LIBRARIES** are needed. So I think the libcurl depends on the nghttp2 package. But you are right these variables are not fillled. What is the recomanded way to these values using the conan generated cmake files?

Reply:
remove this patch, and in `conanfile.py` replace `_patch_cmake` content by:

```python
    def _patch_cmake(self):
        if not self._is_using_cmake_build:
            return
        cmakelists = os.path.join(self._source_subfolder, "CMakeLists.txt")
        # Custom findZstd.cmake file relies on pkg-config file, make sure that it's consumed on all platforms
        if self._has_zstd_option:
            tools.replace_in_file(cmakelists, "if(UNIX)", "if(TRUE)")
        # TODO: check this patch, it's suspicious
        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
                              "include(CurlSymbolHiding)", "")
        # libnghttp2
        tools.replace_in_file(
            cmakelists,
            "find_package(NGHTTP2 REQUIRED)",
            "find_package(libnghttp2 REQUIRED CONFIG)",
        )
        tools.replace_in_file(
            cmakelists,
            "include_directories(${NGHTTP2_INCLUDE_DIRS})",
            "",
        )
        tools.replace_in_file(
            cmakelists,
            "list(APPEND CURL_LIBS ${NGHTTP2_LIBRARIES})",
            "list(APPEND CURL_LIBS libnghttp2::nghttp2)",
        )
```

Reply:
@SpaceIm Thank you for the hint. But unfortunately this doesn't work. After appending libnghttp2::nghttp2 to CURL_LIBS the CheckLibraryExists macro is called. This Marcro delivers the following error:

```-- Looking for idn2_lookup_ul in idn2;winmm;ws2_32;advapi32;OpenSSL::SSL;OpenSSL::Crypto;libnghttp2::nghttp2
CMake Error at R:/.conan/data/libcurl/7.73.0/_/_/build/3ace160ae605b0284e23684536b2df744ddd5e1f/build_subfolder/CMakeFiles/CMakeTmp/CMakeLists.txt:21 (add_executable):
  Target "cmTC_f2cba" links to target
  "CONAN_LIB::libnghttp2_nghttp2_nghttp2DEBUG" but the target was not found.
  Perhaps a find_package() call is missing for an IMPORTED target, or an
  ALIAS target is missing?


CMake Error at C:/Program Files/CMake/share/cmake-3.18/Modules/CheckLibraryExists.cmake:67 (try_compile):
  Failed to generate test project build system.
Call Stack (most recent call first):
  source_subfolder/CMake/Macros.cmake:30 (check_library_exists)
  source_subfolder/CMakeLists.txt:620 (check_library_exists_concat)


-- Configuring incomplete, errors occurred!```

I'm not shure why this happens, as the target schoud be defined in the conan cmake files.